database/control.go

```diff
@@ -61,9 +61,9 @@ func (d *DB) collectionExists(table string) bool {
  * Purpose:  Creates a new collection with required indeces
  * comments:
  */
-func createCollection(d *DB, name string, indeces []string) string {
+func (d *DB) createCollection(name string, indeces []string) string {
```

Comment:
these functions were not object oriented as per golang standards. Just moved the parameter is all